pkgs/development/libraries/science/math/nccl/tests.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, which
+, cudaPackages
+, mpiSupport ? false
+, mpi
+}:
+
+assert mpiSupport -> mpi != null;
+
+with cudaPackages;
+
+cudaPackages.backendStdenv.mkDerivation rec {
```

Comment:
```suggestion
cudaPackages.backendStdenv.mkDerivation (finalAttrs: {
```

Avoid using `rec`; instead, take advantage of the feature introduced in https://github.com/NixOS/nixpkgs/pull/119942 which allows us to give `mkDerivation` a function which can reference itself.